src/svg/qsvggenerator.cpp

```diff
@@ -997,8 +1007,37 @@ void QSvgPaintEngine::updateState(const QPaintEngineState &state)
     // close old state and start a new one...
     if (d->afterFirstUpdate)
         *d->stream << "</g>\n\n";
+    
+    QPainter* p = painter();
+    if (p->hasClipping()) {
+        QPainterPath painter_path = p->clipPathF();
+
+        if (painter_path.elementCount() > 0) {
+            QString clip_path = createPath(painter_path);
+
+            bool path_does_not_exist = 0 == clip_path_to_id.count(clip_path);
+            if (path_does_not_exist) {
+                clip_path_to_id[clip_path] = clip_counter++;
+                
+                d->stream->setString(&d->defs);
+
+                *d->stream << "<clipPath id=\"clip" << clip_path_to_id[clip_path] << "\">" << '\n';
```

Comment:
Is there a reason to use '\n' instead of endl?

Reply:
This is text going directly into the file and I put '\n' to be safe instead of even testing the behaviour of `endl`, I thought it was more readable even if they had the same behaviour.

Reply:
There's a few usages of each in this file, but there's more endl so I'd go with that